fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -111,6 +112,81 @@ public DirectoryReader getWriterReader(boolean flush) throws IOException {
         return writerReader;
     }
 
+    private static class FDBDirectorySerialMergeScheduler extends MergeScheduler {
+        /**
+         * This class is temporarily duplicating FDBDirectoryMergeScheduler.
+         * TODO: After verified, either delete FDBDirectoryMergeScheduler or delete FDBDirectorySerialMergeScheduler.
+         */
+        @Nonnull
+        private final IndexMaintainerState state;
+        private final int mergeDirectoryCount;
+        @Nonnull
+        private final AgilityContext agilityContext;
+        @Nonnull
+        private final Tuple key;
+
+        private FDBDirectorySerialMergeScheduler(@Nonnull IndexMaintainerState state, int mergeDirectoryCount,
+                                           @Nonnull final AgilityContext agilityContext,
+                                           @Nonnull final Tuple key) {
+            this.state = state;
+            this.mergeDirectoryCount = mergeDirectoryCount;
+            this.agilityContext = agilityContext;
+            this.key = key;
+        }
+
+        @SuppressWarnings({
+                "squid:S2245", // ThreadLocalRandom not used in case where cryptographic security is needed
+                "squid:S3776", // Cognitive complexity is too high. Candidate for later refactoring
+        })
+        @Override
+        public synchronized void merge(final MergeSource mergeSource, final MergeTrigger trigger) throws IOException {
+            long startTime = System.nanoTime();
+            if (state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_MULTIPLE_MERGE_OPTIMIZATION_ENABLED) && trigger == MergeTrigger.FULL_FLUSH) {
+                if (ThreadLocalRandom.current().nextInt(mergeDirectoryCount) == 0) {
+                    if (mergeSource.hasPendingMerges()) {
+                        MergeUtils.logExecutingMerge(LOGGER, "Executing Merge based on probability", agilityContext, state.indexSubspace, key, trigger);
```

Comment:
Probably worth differentiating the logs here, from the FDBDirectoryMergeScheduler.
That might be easily done by moving this out so that it is not a nested class, and giving it its own logger.
It might also make sense to change the log of the non-default, but I don't think it's a big deal either way, and suggested edits are easy to apply.
```suggestion
                        MergeUtils.logExecutingMerge(LOGGER, "Executing Merge serially based on probability", agilityContext, state.indexSubspace, key, trigger);
```

Reply:
I strongly agree, but made a slight variation - changing the messages of the twin class to "Executing Merge Concurrently" (as that is the less obvious one). I hope that it's OK with you.